shift_detector/precalculations/DistinctionPrecalculation.py

```diff
@@ -14,10 +15,18 @@
 class DistinctionPrecalculation(Precalculation):
 
     def __init__(self, columns, num_epochs=10):
+        if not isinstance(columns, Iterable) \
+                or any(not isinstance(column, str) for column in columns) \
+                or len(columns) < 1:
```

Comment:
Why can columns be empty in the check but not in the precalculation?

Reply:
Because of the equality / hash of the precalculation. Empty list and all columns defined would calculate the same result but would be impossible to hash on the same value.